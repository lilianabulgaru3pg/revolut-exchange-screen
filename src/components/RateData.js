import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';
import Box from '@material-ui/core/Box';
import PropTypes from 'prop-types';

const useStyles = makeStyles({
  rate: {
    border: `2px solid #E6E6FA`,
    borderRadius: 25,
    width: '30%',
    padding: '5px 8px 4px 8px',
    background: 'white',
  },
});

const RateData = ({ rate }) => {
  const classes = useStyles();
  return (
    <Box className={classes.rate}>
      <Typography color="primary" variant="body2">
        {rate}
      </Typography>
    </Box>
  );
};

RateData.propTypes = {
  rate: PropTypes.string.isRequired,
};

export default RateData;
