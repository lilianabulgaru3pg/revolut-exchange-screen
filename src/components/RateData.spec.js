import React from 'react';
import renderer from 'react-test-renderer';
import RateData from './RateData';

describe('RateData Component', () => {
  test('snapshot renders', () => {
    const component = renderer.create(
      <RateData rate="1$ = 11.00 Â£" />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
