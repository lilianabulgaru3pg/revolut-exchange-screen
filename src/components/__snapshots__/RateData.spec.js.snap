// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RateData Component snapshot renders 1`] = `
<div
  className="MuiBox-root MuiBox-root-3 makeStyles-rate-1"
>
  <p
    className="MuiTypography-root MuiTypography-body2 MuiTypography-colorPrimary"
  >
    1$ = 11.00 Â£
  </p>
</div>
`;
